volunteer/forms.py

```diff
@@ -8,6 +11,120 @@ class VolunteerProfileForm(ModelForm):
+        if discord_username:
+            if not re.match(
+                r"^[a-zA-Z0-9](?:[a-zA-Z0-9]|[._-](?=[a-zA-Z0-9])){0,30}[a-zA-Z0-9]$",
+                discord_username,
+            ):
+                if len(discord_username) < 2 or len(discord_username) > 32:
+                    raise ValidationError(
+                        "Discord username must be between 2 and 32 characters."
+                    )
+                else:
+                    raise ValidationError(
+                        "Discord username must consist of alphanumeric characters, "
+                        "dots, underscores, or hyphens, and cannot have consecutive special characters."
+                    )
+        return discord_username
+
+    def clean_instagram_username(self):
+        instagram_username = self.cleaned_data.get("instagram_username")
+        if instagram_username:
+            if instagram_username.startswith("@"):
+                instagram_username = instagram_username[1:]
+
+            if not re.match(r"^[a-zA-Z0-9._]{1,30}$", instagram_username):
+                raise ValidationError(
+                    "Instagram username can only contain alphanumeric characters, "
+                    "periods, and underscores, and must be between 1-30 characters."
+                )
+        return instagram_username
+
+    def clean_bluesky_username(self):
+        bluesky_username = self.cleaned_data.get("bluesky_username")
+        if bluesky_username:
+            if bluesky_username.startswith("@"):
+                bluesky_username = bluesky_username[1:]
+
+            if not re.match(
+                r"^[a-zA-Z0-9][a-zA-Z0-9.-]{0,28}[a-zA-Z0-9](\.[a-zA-Z0-9][\w.-]*\.[a-zA-Z]{2,})?$",
+                bluesky_username,
+            ):
+                raise ValidationError(
+                    "Invalid Bluesky username format. "
+                    "Should be either a simple username or a full handle (e.g., username.bsky.social)."
+                )
+        return bluesky_username
+
+    def clean_mastodon_url(self):
+        mastodon_url = self.cleaned_data.get("mastodon_url")
+        if mastodon_url:
+            mastodon_pattern1 = r"^@[a-zA-Z0-9_]+@[a-zA-Z0-9][a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$"  # @user@instance.tld
+            mastodon_pattern2 = r"^https?://[a-zA-Z0-9][a-zA-Z0-9.-]+\.[a-zA-Z]{2,}/@[a-zA-Z0-9_]+$"  # https://instance.tld/@user
+
+            if not (
+                re.match(mastodon_pattern1, mastodon_url)
+                or re.match(mastodon_pattern2, mastodon_url)
+            ):
+                raise ValidationError(
+                    "Invalid Mastodon URL format. "
+                    "Should be either @username@instance.tld or https://instance.tld/@username."
+                )
+        return mastodon_url
+
+    def clean_x_username(self):
+        x_username = self.cleaned_data.get("x_username")
+        if x_username:
+            if x_username.startswith("@"):
```

Comment:
Same as for instagram and bluesky, it should raise validation error instead of silently stripping @ from the username